src/util/Algorithm.h

```diff
@@ -108,6 +110,23 @@ std::vector<T> flatten(std::vector<std::vector<T>>&& input) {
   return out;
 }
 
+/**
+ * Remove duplicates in the given vector without chaning the order. For example:
+ * 4, 6, 6, 2, 2, 4, 2 becomes 4, 6, 2.
+ */
+template <typename T>
+std::vector<T> removeDuplicates(std::vector<T>&& input) {
+  std::vector<T> result;
+  ad_utility::HashSet<T> distinctElements;
+  for (const T& element : input) {
+    if (!distinctElements.contains(element)) {
+      result.emplace_back(element);
+      distinctElements.insert(element);
```

Comment:
That would be weird because then only some elements would be moved (the first occurence of each kind). I now pass the input as const&, then at least it's easier to understand. I added a comment that this is making a lot of copies.